### PR TITLE
docs(localforage): js/coffee -> promises/callbacks

### LINKED ISSUE #534: Docs improvement - callbacks/promise switcher
There was a thread on Twitter recently from @domenic about [newbies' difficulty using LocalForage](https://twitter.com/domenic/status/706220942096257024). I think it's a bit sad he was willing to look for alternatives, when really LocalForage ought to be a very beginner-friendly API. I suspect the real issue is a docs problem.

I have a couple suggestions, but I wanted to run them past @tofumatt first to get his thoughts:

1. **Replace the ["javascript/coffeescript" switch](http://mozilla.github.io/localForage/) in the docs with a "promise/callback" switch.** Coffeescript users are typically power users, so IME they don't need any extra documentation. Also the promise/callback switch idea was something we used [in the PouchDB docs](http://pouchdb.com/api.html#create_document), and I think the effect is really nice.
2. **Remove any documentation from the Github README.** Seriously, this can only serve to confuse people. Newbies aren't sure whether the "real" docs are in the README or on the site. Just make the README essentially a stub; point people to [the docs page](http://mozilla.github.io/localForage/).

### PR DESCRIPTION
**Pull request by nolanlawson**
Fixes #534.

### CODE REVIEW

**Comment by nolanlawson** [comment]
I also removed most of the promise/callback explanation from the README, directing people to the docs instead.

Since the old js/coffee switch was kept in LocalStorage, I changed the LocalStorage key. The effect is that your previous js/coffee decision doesn't matter anymore; promises are the new default, although you can switch to callbacks if you really like.

Note that the even/odd switch is a bit hacky, but I couldn't think of a better way to do this without directly injecting HTML in the docs as opposed to markdown.

Also note that I added `.catch()` everywhere in the docs. I figure it's better to insulate people from uncaught errors at the expense of a little noisiness in the docs. (We did the same in the Pouch docs FWIW).

**Comment by thgreasi** [comment]
+2!!!

**Comment by thgreasi** [comment]
Why there is change on the dist js files? (Possibly a newer version of the minifier was released!?!?)

File: README.md [hunk 1]
```diff
@@ -48,82 +48,39 @@ is usually online from 8am-8pm GMT (London Time).
...
+  // if err is non-null, we got an error
+  localforage.getItem('key', function (err, value) {
+    // if err is non-null, we got an error. otherwise, value is the value 
+  });
 });
 ```
 
-You can store any type in localForage; you aren't limited to strings like in
-localStorage. Even if localStorage is your storage backend, localForage
-automatically does `JSON.parse()` and `JSON.stringify()` when getting/setting
```
Comment (thgreasi) [inline README.md]: How about moving this paragraph right after [line 55](https://github.com/mozilla/localForage/pull/536/files#diff-04c6e90faac2675aa89e2176d2eec7d8R55)?

**Comment by nolanlawson** [comment]
> Why there is change on the dist js files?

No idea. Feel free to remove that on the PR when you merge, though. But yeah, I suspect it's because of a dep update.

File: README.md (hunk 1 above)
Comment (nolanlawson) [inline README.md]: Yes, I didn't mean to remove that. Good catch!

**Comment by nolanlawson** [comment]
OK, removed the `dist` files and updated the README as requested.

**Comment by nolanlawson** [comment]
Ping @thgreasi @tofumatt, can we get this merged to master/gh-pages? I really think it'd be great to have better docs published for LocalForage, so we can show off how simple and easy-to-use the API is. :smiley:

**Comment by tofumatt** [comment]
I've been away on holiday but I'll do it soon. I just landed in London from
Beijing, so I'll probably look at this tomorrow if I'm being totally honest
:-)
- tofumatt

**Comment by nolanlawson** [comment]
Oh OK no rush. :)

**Comment by tofumatt** [comment]
I'm having some issues building the docs but it is a priority to get this online. I just came back from holidays, have a work week in Berlin this week, and am in the early stages of buying at house. _Slightly_ busy, but I am on it!

**Pull request merged**